lib/util/chat/message_bar.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../supabase.dart';
+
+class MessageBar extends StatefulWidget {
+  const MessageBar(this.chatId, {super.key});
+  final int chatId;
+
+  @override
+  State<MessageBar> createState() => _MessageBarState();
+}
+
+class _MessageBarState extends State<MessageBar> {
+  late final TextEditingController _textController;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(0, 20, 0, 0),
+      child: Material(
+        color: Theme.of(context).colorScheme.onSurface.withOpacity(0.1),
+        child: SafeArea(
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Row(
+              children: <Widget>[
+                Expanded(
+                  child: TextFormField(
+                    keyboardType: TextInputType.text,
+                    maxLines: null,
+                    autofocus: true,
+                    controller: _textController,
+                    decoration: InputDecoration(
+                      hintText: S.of(context).pageChatMessageBarHint,
+                      border: InputBorder.none,
+                      focusedBorder: InputBorder.none,
+                      contentPadding: const EdgeInsets.all(8),
+                    ),
+                  ),
+                ),
+                IconButton(
+                  onPressed: () => _submitMessage(),
+                  icon: const Icon(Icons.send),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  void initState() {
+    _textController = TextEditingController();
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _textController.dispose();
+    super.dispose();
+  }
+
+  Future<void> _submitMessage() async {
+    final String text = _textController.text;
+    if (text.isEmpty) {
+      return;
+    }
+    _textController.clear();
+    try {
+      await SupabaseManager.supabaseClient.from('messages').insert(<String, dynamic>{
+        'sender_id': SupabaseManager.getCurrentProfile()!.id,
+        'content': text,
+        'chat_id': widget.chatId,
+      });
+    } on Exception {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text(S.of(context).failureSnackBar),
```

Comment:
Every time I try to send a message as rider, this snackbar appears. And I don't see any messages. Maybe actually something wrong with RLS? It worked for Simon for some reason.

Reply:
the RLS is off that can't be the reason for me it works fine so not sure what's happening there

Reply:
you can try an show the Exception

Reply:
flutter: PostgrestException(message: insert or update on table "messages" violates foreign key constraint "messages_chat_id_fkey", code: 23503, details: Key (chat_id)=(-1) is not present in table "chats"., hint: null)

Reply:
thanls that helps

Reply:
is it possible that there is no chat for your ride?

Reply:
the chat_id should only be -1 if there is no chat

Reply:
No I just checked and there is a corresponding chat in supabase

Reply:
that's weird then the id should not be -1.

Reply:
Maybe the supabase trigger just takes too long?

Reply:
no that shouldn't take long

Reply:
for me it's working i don't understand what's happening. When exactly did that happen?

Reply:
I can't reproduce it anymore either... It's really worrying that this is so flaky, but I'm closing this for now